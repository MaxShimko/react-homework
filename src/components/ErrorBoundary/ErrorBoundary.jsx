import React from 'react';

class ErrorBoundary extends React.Component {
    constructor(props) {
        super(props);
        this.state = { hasError: false };
    }

    static getDerivedStateFromError(error) {
        return { hasError: true };
    }

    componentDidCatch(error, errorInfo) {
        console.log(error, errorInfo);
    }

    render() {
        if (this.state.hasError) {
            return (
                <div style={{display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100vh',
                    backgroundColor: '#424242', color: '#F65261'}}>
                    <h1>Something went wrong.</h1>
                </div>
            );
        }

        return this.props.children;
    }
}

export default ErrorBoundary;
