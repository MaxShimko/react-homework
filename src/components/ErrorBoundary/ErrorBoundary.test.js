import React from 'react';
import {shallow} from 'enzyme';
import ErrorBoundary from './ErrorBoundary';
import {describe, expect, it} from '@jest/globals';
import toJson from 'enzyme-to-json';

describe('<ErrorBoundary />', () => {
    it('renders without crashing', () => {
        const component = shallow(<ErrorBoundary/>);
        expect(toJson(component)).toMatchSnapshot();
    });
});
