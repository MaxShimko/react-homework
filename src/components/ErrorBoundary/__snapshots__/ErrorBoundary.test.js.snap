// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> renders without crashing 1`] = `""`;
